OptimoveSDK/Sources/Classes/PreferenceCenter/OptimovePreferenceCenter.swift

```diff
@@ -0,0 +1,197 @@
+        }
+
+        instance = OptimovePreferenceCenter(storage: storage, networkClient: networkClient)
+    }
+
+    private init(storage: OptimoveStorage, networkClient: NetworkClient) {
+        self.networkClient = networkClient
+        self.storage = storage
+    }
+
+    public func getPreferencesAsync(completion: @escaping PreferencesGetHandler) {
+        guard let config = Optimove.getConfig()?.getPreferenceCenterConfig() else {
+            Logger.error("Preference center credentials are not set")
+            completion(.errorNotConfigured, nil)
+            return
+        }
+
+        guard
+            let customerId = try? storage?.getCustomerID(),
+            let visitorId = try? storage?.getVisitorID(),
+            customerId != visitorId
+        else {
+            Logger.warn("Customer ID is not set")
+            completion(.errorUserNotSet, nil)
+            return
+        }
+
+        Task {
+            do {
+                let request = try createGetPreferencesRequest(for: customerId, with: config)
+                guard let data = try await networkClient?.performAsync(request) else {
+                    throw NetworkError.noData
+                }
+                let preferences = try JSONDecoder().decode(OptimovePC.Preferences.self, from: data)
+
+                DispatchQueue.main.async {
+                    completion(.success, preferences)
+                }
+            } catch {
+                logFailedResponse(error)
+                DispatchQueue.main.async {
+                    completion(.error, nil)
+                }
+            }
+        }
+    }
+    
+    private func createGetPreferencesRequest(
+        for customerId: String,
+        with config: PreferenceCenterConfig) throws -> NetworkRequest {
+            let (region, brandGroupId, tenantId) = try getConfigValues(from: config)
+
+            return NetworkRequest(
+                method: .get,
+                baseURL: URL(string: "https://preference-center-\(region).optimove.net")!,
+                path: "/api/v1/preferences",
+                headers: [
+                    HTTPHeader(field: .accept, value: .textplain),
+                    HTTPHeader(field: .tenantId, value: .tenantId(id: tenantId))
+                ],
+                queryItems: [
+                    URLQueryItem(name: "customerId", value: customerId),
+                    URLQueryItem(name: "brandGroupId", value: brandGroupId)
+                ]
+            )
+    }
+
+    private func getConfigValues(from config: PreferenceCenterConfig) throws -> (region: String, brandGroupId: String, tenantId: String) {
+        let region = config.region
+        let brandGroupId = config.brandGroupId
+        let tenantId = config.tenantId.description
+        return (region, brandGroupId, tenantId)
+    }
+
+
+    public func setPreferencesAsync(updates: [OptimovePC.PreferenceUpdate], completion: @escaping PreferencesSetHandler) {
```

Comment:
one from android is `public void setCustomerPreferencesAsync(@NonNull PreferencesSetHandler preferencesSetHandler, List<PreferenceUpdate> updates) {`

1. arg order
2. diff name

Reply:
@vvoicehovics Android has
- `getPreferencesAsync`
- `setCustomerPreferencesAsync`

Should we update it to `getCustomerPreferencesAsync`

Reply:
suppose no, it was intentional -- back to that Preferences ambiguity

- `getPreferences` gives you customer preferences, but also some PC configuration. Later it may give you more random stuff, we just get some ill-defined "Preferences"
- `setCustomerPreferences` highlights that we are setting only preferences for the customer, nothing else

web and android do same. Reasoning dubious as usual :) but let's keep consistent with rest